### DISCUSSION TITLE
Do I need to stop udisks2 for initrd flash?

### DISCUSSION THREAD

**Discussion by CHNtentes** [OE4T/meta-tegra#1181] [Q&A] [answered]
According to NVIDIA's README file of initrd flash, I need to run `systemctl stop udisks2.service` before I do a initrd flash.
Is this the same with meta-tegra solution?

**Comment by madisongh** [comment, marked as answer]
No, you don't have to disable udisks2.  In fact, as mentioned on [wiki page](https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support), the scripts use `udisksctl` (which talks to udisks2) to manage the USB mass storage devices.

And from my testing, I don't think you actually need to with stock L4T, either, at least with recent versions.

**Reply by CHNtentes** [reply]
Hi. When I run initrd-flash today, it gives me some error:

> Error mounting /dev/sdb: GDBus.Error:org.freedesktop.UDisks2.Error.AlreadyMounted: Device /dev/sdb is already mounted at `/media/v2x/6c14c7cf-61c6-4481-a576-36c8f391f7be'.

What could be the reason for this issue? I already enabled "Never prompt or start programs on media insertion" in system settings.

The logs are here:
[log.initrd-flash.2023-02-24-16.48.11.txt](https://github.com/OE4T/meta-tegra/files/10822343/log.initrd-flash.2023-02-24-16.48.11.txt)
[uart_log.txt](https://github.com/OE4T/meta-tegra/files/10822345/uart_log.txt)

**Reply by madisongh** [reply]
Sounds like the automount behavior did not get shut off after all.

Can you check
```
$ dconf read /org/gnome/desktop/media-handling/automount
```
If that's still set to `true`, then try
```
$ dconf write /org/gnome/desktop/media-handling/automount false
```
and see if the problem goes away?  If so, I'll update the wiki page to mention this.

**Reply by CHNtentes** [reply]
Sorry for late reply. I tried reading the value, but strangely it did not return anything. Then I wrote false to it and flashed again, the automount error was gone and flash succeeded. So I guess this is the fix. Thanks a lot.

**Reply by pkgl** [reply]
Just came around this error as well. Just figured out if you are using a remote session you will need to start dbus-launch to be able to edit settings with dconf.

```
$ dbus-launch dconf read /org/gnome/desktop/media-handling/automount
$ dbus-launch dconf write /org/gnome/desktop/media-handling/automount false
```
